R/env.R

```diff
@@ -0,0 +1,80 @@
+
+load_env <- function(path, targz, package, envir = parent.frame()) {
+  should_load <- as_flag(
+    Sys.getenv("RCMDCHECK_LOAD_CHECK_ENV"),
+    as_flag(Sys.getenv("NOT_CRAN"))
+  )
+  if (!should_load) return()
+
+  env <- NULL
+  if (file.info(path)$isdir) {
+    env_path <- file.path(path, "inst", "check.env")
+  } else {
+    dir.create(tmp <- tempfile())
+    on.exit(unlink(tmp, recursive = TRUE), add = TRUE)
+    untar(
+      targz,
+      file.path(package, "inst", "check.env"),
+      exdir = tmp, tar = "internal"
+    )
+    env_path <- file.path(tmp, package, "inst", "check.env")
+  }
+
+  if (file.exists(env_path)) {
+    load_env_file(env_path, envir = envir)
+  }
+}
+
+load_env_file <- function(path, envir = parent.frame()) {
```

Comment:
Should this use `readRenviron()` instead of using our own implementation?

Reply:
> I guess the downside would be restoring the old envvars afterwards. But you could do Sys.getenv() to save them beforehand I guess.

I don't really like its way of handling errors, it just ignores them, and especially on a CI, it is better to know about them.

So it was easier to copy this code from another package of mine. 

But I don't really mind, we can switch to `readRenviron()` as well.